tests/test_prepare_docs.py

```
@@ -0,0 +1,130 @@
+        os.makedirs("docs", exist_ok=True)
+        self._write("docs/doc1.md", "# First Doc\nContent")
+        self._write("docs/doc2.md", "---\ntitle: \"Second\"\n---\nContent")
+        self._write("docs/ignored.txt", "Should not be touched")
+
+        # Mock os.path.abspath to return the path under our temp directory for 'docs'
+        real_docs_path = os.path.abspath(os.path.join(REPO_ROOT, "docs"))
+        temp_docs_path = os.path.abspath("docs")
+
+        orig_abspath = os.path.abspath
+
+        def mock_abspath(path):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `mock_abspath`

(ANN202)